BuildResidentialHPXML/measure.rb

```diff
@@ -4725,51 +4243,23 @@ def self.set_hvac_distribution(hpxml, runner, args)
     end
   end
 
-  def self.modify_setpoint_schedule(schedule, offset_magnitude, offset_schedule)
-    offset_schedule.each_with_index do |direction, i|
-      schedule[i] += offset_magnitude * direction
-    end
-    return schedule
-  end
-
   def self.set_hvac_control(hpxml, runner, args)
     return if (args[:heating_system_type] == 'none') && (args[:cooling_system_type] == 'none') && (args[:heat_pump_type] == 'none')
 
-    weekday_heating_setpoints = [args[:setpoint_heating_weekday_temp]] * 24
-    weekend_heating_setpoints = [args[:setpoint_heating_weekend_temp]] * 24
-
-    weekday_cooling_setpoints = [args[:setpoint_cooling_weekday_temp]] * 24
-    weekend_cooling_setpoints = [args[:setpoint_cooling_weekend_temp]] * 24
-
-    if args[:setpoint_heating_weekday_offset_magnitude].is_initialized && args[:setpoint_heating_weekday_schedule].is_initialized
-      setpoint_heating_weekday_offset_magnitude = args[:setpoint_heating_weekday_offset_magnitude].get
-      setpoint_heating_weekday_schedule = args[:setpoint_heating_weekday_schedule].get.split(',').map { |i| Float(i) }
-      weekday_heating_setpoints = modify_setpoint_schedule(weekday_heating_setpoints, setpoint_heating_weekday_offset_magnitude, setpoint_heating_weekday_schedule)
-    end
-
-    if args[:setpoint_heating_weekend_offset_magnitude].is_initialized && args[:setpoint_heating_weekend_schedule].is_initialized
-      setpoint_heating_weekend_offset_magnitude = args[:setpoint_heating_weekend_offset_magnitude].get
-      setpoint_heating_weekend_schedule = args[:setpoint_heating_weekend_schedule].get.split(',').map { |i| Float(i) }
-      weekend_heating_setpoints = modify_setpoint_schedule(weekend_heating_setpoints, setpoint_heating_weekend_offset_magnitude, setpoint_heating_weekend_schedule)
-    end
-
-    if args[:setpoint_cooling_weekday_offset_magnitude].is_initialized && args[:setpoint_cooling_weekday_schedule].is_initialized
-      setpoint_cooling_weekday_offset_magnitude = args[:setpoint_cooling_weekday_offset_magnitude].get
-      setpoint_cooling_weekday_schedule = args[:setpoint_cooling_weekday_schedule].get.split(',').map { |i| Float(i) }
-      weekday_cooling_setpoints = modify_setpoint_schedule(weekday_cooling_setpoints, setpoint_cooling_weekday_offset_magnitude, setpoint_cooling_weekday_schedule)
+    if args[:setpoint_heating_weekday] == args[:setpoint_heating_weekend] && !args[:setpoint_heating_weekday].include?(',')
+      heating_setpoint_temp = args[:setpoint_heating_weekday]
```

Comment:
So if they enter a single value and it's the same for weekdays and weekends, it gets written to the standard HPXML `SetpointTempHeatingSeason` and `SetpointTempCoolingSeason` elements right? And if not, it uses the 24-hour extension elements? Big 👍 if so.

Reply:
That's the intent, yes.